go/logic/applier.go

```diff
@@ -753,6 +757,35 @@ func (this *Applier) ApplyIterationInsertQuery() (chunkSize int64, rowsAffected
 		if err != nil {
 			return nil, err
 		}
+
+		if this.migrationContext.PanicOnWarnings {
+			//nolint:execinquery
+			rows, err := tx.Query("SHOW WARNINGS")
+			if err != nil {
+				return nil, err
+			}
+			defer rows.Close()
+			if err = rows.Err(); err != nil {
+				return nil, err
+			}
+
+			var sqlWarnings []string
+			for rows.Next() {
+				var level, message string
+				var code int
+				if err := rows.Scan(&level, &code, &message); err != nil {
+					this.migrationContext.Log.Warningf("Failed to read SHOW WARNINGS row")
+					continue
+				}
+				pkDuplicateSuffix := fmt.Sprintf("for key '%s.PRIMARY'", this.migrationContext.GetGhostTableName())
+				if strings.HasPrefix(message, "Duplicate entry") && strings.HasSuffix(message, pkDuplicateSuffix) {
```

Comment:
We may have a different shared unique key than the primary key (`migrationContext.UniqueKey`)

Reply:
Good catch! Based on how the shared key is selected in `inspectOriginalAndGhostTables`, it seems the correct behavior here would be to ignore any warnings including `migrationContext.UniqueKey.Name` instead of just hardcoding the pk. This is a logic spill from LHM - I'll fix that and see if I can add a unit test.

Reply:
Addressed in [36eaf27](https://github.com/github/gh-ost/pull/1500/commits/36eaf27e4c7f0a425ebd7192d18ff5e885744587) 👍